hoomd/mesh.py

```diff
@@ -102,13 +105,33 @@ def triangles(self):
     @triangles.setter
     def triangles(self, triag):
         if self._attached:
-            self._cpp_obj.setTypes(list(self._param_dict['types']))
 
-            self._cpp_obj.setTriangleData(triag)
+            if len(triag) != len(self._type_ids):
+                raise ValueError(
+                    "Number of type_ids do not match number of triangles.")
+            else:
+                self._cpp_obj.setTypes(list(self._param_dict['types']))
+
+                self._cpp_obj.setTriangleData(triag, self._type_ids)
         else:
             self.size = len(triag)
         self._triangles = triag
 
+    @log(category='sequence')
+    def type_ids(self):
+        """((*N*) `numpy.ndarray` of ``uint32``): Triangle type ids."""
+        if self._attached:
+            return self._cpp_obj.getTriangleData().typeid
+        return self._type_ids
+
+    @type_ids.setter
+    def type_ids(self, tid):
+        if self._attached and self.size == len(tid):
```

Comment:
Why is the second condition necessary?

Reply:
Here I check if the number of triangles is equal the number of type_ids